Source/Engine/Utilities/Extensions.cs

```diff
@@ -404,5 +238,184 @@ public static void Shuffle<T>(this Random random, IList<T> collection)
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <param name="randomRoll">Should the roll value be randomized.</param>
+        /// <returns>A random <see cref="Quaternion"/>.</returns>
+        public static Quaternion NextQuaternion(this Random random, bool randomRoll = false)
+        {
+
+            return Quaternion.Euler(
+                NextFloat(random, -180, 180),
+                NextFloat(random, -180, 180),
+                randomRoll ? NextFloat(random, -180, 180) : 0);
+        }
+
+        /// <summary>
+        /// Generates a uniformly distributed random unit length vector point on a unit sphere.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <returns>A random <see cref="Vector3"/>.</returns>
+        public static Vector3 NextVector3(this Random random)
+        {
+            Vector3 output;
+
+            do
+            {
+                // Create random float with a mean of 0 and deviation of ±1
+                output.X = NextFloat(random) * 2.0f - 1.0f;
+                output.Y = NextFloat(random) * 2.0f - 1.0f;
+                output.Z = NextFloat(random) * 2.0f - 1.0f;
+
+            } while (output.LengthSquared > 1 || output.LengthSquared < 1e-6f);
+
+            output *= (1.0f / (float)Math.Sqrt(output.LengthSquared));
+
+            return output;
+        }
+
+        /// <summary>
+        /// Generates a uniformly distributed random unit length vector point on a unit sphere in 2D.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <returns>A random <see cref="Vector2"/>.</returns>
+        public static Vector2 NextVector2(this Random random)
+        {
+            Vector2 output;
+
+            do
+            {
+                output.X = NextFloat(random) * 2.0f - 1.0f;
+                output.Y = NextFloat(random) * 2.0f - 1.0f;
+
+            } while (output.LengthSquared > 1 || output.LengthSquared < 1e-6f);
+
+            output *= (1.0f / (float)Math.Sqrt(output.LengthSquared));
+
+            return output;
+        }
+
+        /// <summary>
+        /// Generates a random <see cref="Color"/>.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <param name="trueRandom">Should the color be generated from a single random Hue value or separate values for each channel.</param>
+        /// <param name="randomAlpha">Randomize the alpha value.</param>
+        /// <returns>A nice random <see cref="Color"/>.</returns>
+        public static Color NextColor(this Random random, bool trueRandom, bool randomAlpha)
+        {
+            float alpha = randomAlpha ? NextFloat(random) : 1f;
+
+            if (!trueRandom)
+                return Color.FromHSV(NextFloat(random, 0f, 360f), 1f, 1f, alpha);
```

Comment:
I'm torn about this one, it's super useful but it might almost deserve its own method?

Reply:
I'm not sure about it either, It can't be an overloaded method because then the parameters of those 2 functions would be equal as both would have the randomize alpha param.

So would it be worth it making a whole new method just for selecting from HSV instead?

Reply:
I guess it might be worthwhile to make a whole new method for a random HSV color.

Reply:
Done.